sunpy/coordinates/utils.py

```diff
@@ -265,3 +265,69 @@ def coordinates(self, points=None):
+
+
+def get_rectangle_coordinates(bottom_left, *, top_right = None, width: u.deg = None, height: u.deg = None):
+    """
+    Specify a rectangular region of interest in spherical world coordinates.
+
+    Parameters
+    ----------
+    bottom_left : `~astropy.coordinates.BaseCoordinateFrame` or `~astropy.coordinates.SkyCoord`
+        The bottom_left coordinate of the rectangle. If a `SkyCoord` it can
+        have shape ``(2,)`` and also define ``top_right``.
+
+    top_right : `~astropy.coordinates.BaseCoordinateFrame` or `~astropy.coordinates.SkyCoord`
+        The top_right coordinate of the rectangle. Can only be omitted if
+        ``bottom_left`` has shape ``(2,)``.
+
+    width : `~astropy.units.Quantity`
+        The width of the rectangle.
+        Must be omitted if ``bottom_left`` has shape ``(2,)`` or ``top_right`` is provided.
+
+    height : `~astropy.units.Quantity`
+        The height of the rectangle.
+        Must be omitted if ``bottom_left`` has shape ``(2,)`` or ``top_right`` is provided.
+
+    Returns
+    -------
+    bottom_left : `~astropy.coordinates.BaseCoordinateFrame` or `~astropy.coordinates.SkyCoord`
+        The bottom_left coordinate of the rectangular region of interest in spherical world coordinates.
+
+    top_right : `~astropy.coordinates.BaseCoordinateFrame` or `~astropy.coordinates.SkyCoord`    
+        The top_right coordinate of the rectangular region of interest in spherical world coordinates.
+    """
+    if bottom_left.shape == (2,) and any((width, height, top_right)):
+        raise ValueError("Invalid input, if bottom_left.shape == (2,) \
+                          other parameters should not be passed.")
+
+    if all((width, height, top_right)):
+        raise ValueError("Invalid input, width, height and top_right \
+                          parameters should not be passed simultaneously.")
+
+    if top_right is None and bottom_left.shape != (2,) and not all((width, height)):
+        raise ValueError("Invalid input, either bottom_left and top_right \
+                         or bottom_left and height and width should be provided.")
+
+    if (not isinstance(bottom_left, (SkyCoord, BaseCoordinateFrame)) and 
+        not isinstance(top_right, (None, SkyCoord, BaseCoordinateFrame))):
+        raise TypeError("bottom_left and top_right must be of type SkyCoord or BaseCoordinateFrame")
+
+    elif bottom_left.shape == (2,):
+        top_right = bottom_left[1]
+        bottom_left = bottom_left[0]
+
+    elif top_right is not None:
+        top_right = top_right.transform_to(bottom_left.frame)        
+        width = top_right.spherical.lon - bottom_left.spherical.lon
+        height = top_right.spherical.lat - bottom_left.spherical.lat
+
+        if width <= 0 * u.deg or height <= 0 * u.deg:
+            raise ValueError("top_right must be to the right and above bottom_left.")
+
+    else:
+        top_right = SkyCoord(bottom_left.spherical.lon + width,
```

Comment:
So, `top_right` will always be a SkyCoord, even if `bottom_left` is supplied as a frame instance?  That's a bit weird.

Reply:
yeah, this should be conditional on the input type.
I think you can do
```
top_right = bottom_left.realize_frame(...)
if isinstance(bottom_left, SkyCoord):
    top_right = SkyCoord(top_right.data, frame=bottom_left)
```
not sure if that's the best way to do it though, @ayshih ?

Reply:
What I'm doing right now is basically this but in opposite order.
top_right becomes a SkyCoord first  and if bottom_left is a frame, I downgrade it to the corresponding frame.
Would the opposite order be more efficient?

Reply:
It's fine either way.  The SkyCoord constructor is a little additional overhead, but I'd argue that it's more readable.  Adding a comment to the downgrading would be prudent.

(If you change it to @Cadair's way, use `SkyCoord(top_right)`.)